ntp-proto/src/algorithm/mod.rs

```diff
@@ -18,6 +18,29 @@ pub struct ObservablePeerTimedata {
     pub last_update: NtpTimestamp,
 }
 
+#[derive(Debug, Clone)]
+pub struct StateUpdate<PeerID: Eq + Copy + Debug> {
```

Comment:
By whom? I am not familiar with this convention and personally prefer to have them on the datatype as well to make it conceptually clear that this data type doesn't make sense without these bounds.

Reply:
Sorry, but I am still missing the rationale for this. The standard library has a number of conventions specifically to do with backwards compatibility, and as far as I know this might just be one of those. Backwards compatibility isn't applicable that way in our context so I am a bit loathe to just take this convention because "the standard library does it that way."